### PR TITLE
(src) LNMTL & Scribblehub

### PR DESCRIPTION
**Pull request by K1ngfish3r**
added filters
added icon
added scribblehub cause i used wrong branch as ref when making new branch oops

### CODE REVIEW

File: plugins/english/lnmtl.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+        link += `orderBy=${filters.order.value}`;
+        link += `&order=${filters.sort.value}`;
+        link += `&filter=${filters.storyStatus.value}`;
+        link += `&page=${page}`;                         
+
+        const headers = new Headers();
+        const body = await fetchApi(link, { headers }).then((result) =>
+            result.text()
+        );
```
Comment (nyagami) [inline plugins/english/lnmtl.ts:27]:
```suggestion
        const body = await fetchApi(link).then((result) =>
            result.text()
        );
```

File: plugins/english/lnmtl.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            novels.push(novel);
+        });
+
+        return novels;
+    };
+
+    async parseNovelAndChapters(novelUrl: string): Promise<Plugin.SourceNovel> {
+        const url = novelUrl;
+        const headers = new Headers();
+        const result = await fetchApi(url, { headers });
```
Comment (nyagami) [inline plugins/english/lnmtl.ts:54]:
```suggestion
        const result = await fetchApi(url);
```

File: plugins/english/lnmtl.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                    novel.status = detail;
+                    break;
+            }
+        });
+
+        novel.genres = loadedCheerio('.panel-heading:contains(" Genres ")')
+            .next()
+            .text()
+            .trim()
+            .replace(/\s\s/g, ",");
```
Comment (nyagami) [inline plugins/english/lnmtl.ts:88]: are you sure this correct?
what there's novel with genres: `Yuri` `Comedy` `Adapted to Anime`

File: plugins/english/lnmtl.ts [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+            const volumePage = await volumeData.json();
+
+            // volumeData = volumeData.data.map((volume) => volume.slug);
+
+            for (let i = 1; i <= volumePage.last_page; i++) {
+                const chapterData = await fetchApi(
+                    `${this.site}chapter?page=${i}&volumeId=${volume}`
+                );
+                const chapterInfo = await chapterData.json();
+
```
Comment (nyagami) [inline plugins/english/lnmtl.ts:115]: should have a delay

File: plugins/english/lnmtl.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        }
+
+        novel.chapters = chapter;
+
+        return novel;
+    };
+
+    async parseChapter(chapterUrl: string): Promise<string> {
+        const headers = new Headers();
+        const result = await fetchApi(chapterUrl, { headers });
```
Comment (nyagami) [inline plugins/english/lnmtl.ts:134]:
```suggestion
        const result = await fetchApi(chapterUrl);
```

**Review by nyagami — CHANGES_REQUESTED** [review]

File: plugins/english/lnmtl.ts (hunk 3 above)
Comment (K1ngfish3r) [reply plugins/english/lnmtl.ts:88]: output from this command for Cultivation Chat Group https://lnmtl.com/novel/cultivation-chat-group
``"genres": "Action,Adventure,Comedy,Fantasy,Martial Arts,Mystery,School Life,Slice of Life,Xianxia"``

File: plugins/english/lnmtl.ts (hunk 4 above)
Comment (K1ngfish3r) [reply plugins/english/lnmtl.ts:115]: added delay of 1000ms

File: plugins/english/lnmtl.ts (hunk 3 above)
Comment (K1ngfish3r) [reply plugins/english/lnmtl.ts]: updated code

File: plugins/english/scribblehub.ts [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+
+        loadedCheerio('.toc_w').each(function () {
+            const chapterName = loadedCheerio(this).find('.toc_a').text();
+            const releaseDate = loadedCheerio(this).find('.fic_date_pub').text();
+            const chapterUrl = loadedCheerio(this).find('a').attr('href');
+
+            if (!chapterUrl) return;
+            chapter.push({
+                name: chapterName,
+                releaseTime: parseMadaraDate(releaseDate),
```
Comment (nyagami) [inline plugins/english/scribblehub.ts:135]: parseMadaraDate is not used for this case. I think I will remove it from lib

Comment (nyagami) [reply plugins/english/scribblehub.ts:135]: convert it into iso string instead

Comment (K1ngfish3r) [reply plugins/english/scribblehub.ts]: i have copypasta parseMadaraDate into .ts

**Pull request closed without merging**